mmv1/third_party/terraform/services/dataflow/resource_dataflow_flex_template_job.go.erb

```diff
@@ -364,7 +364,16 @@ func resourceDataflowFlexJobSetupEnv(d *schema.ResourceData, config *transport_t
 	var enableStreamingEngine bool
 	if p, ok := d.GetOk("parameters.enableStreamingEngine"); ok {
 		delete(updatedParameters, "enableStreamingEngine")
-		enableStreamingEngine = p.(bool)
+		e := strings.ToLower(p.(string))
+		switch e {
+		case "true":
+			enableStreamingEngine = true
+		case "false":
+			enableStreamingEngine = false
+		default:
+			log.Printf("[DEBUG] Map entry `parameters.enableStreamingEngine` has unexpected value `%s`. It should be either \"true\" or \"false\". Defaulting to false boolean value instead.", e)
```

Comment:
Would it not be better to actually return this as a full failure (even if for debug purposes only) -- because these fields are immutable, if somehow a user triggered this (unlikely) scenario while intending the other outcome, they would then require a full recreation

Reply:
[15dbe03](https://github.com/GoogleCloudPlatform/magic-modules/pull/10303/commits/15dbe0344d8df5fd3cffe122ab06413b210a2210)